distro/otel_test.go

```diff
@@ -611,54 +644,64 @@ func (coll *collector) Start(t *testing.T) {
 	require.NoError(t, conn.Close())
 }
 
-func (coll *collector) SpansExportRequest() *spansExportRequest {
-	// Give some time for the gRPC server to process the request.
-	timeout := time.NewTimer(100 * time.Millisecond)
-	defer timeout.Stop()
-	select {
-	case got := <-coll.traceService.requests:
-		return &got
-	case <-timeout.C:
-		return nil
-	}
+func (coll *collector) ExportedSpans() *spansExportRequest {
+	// stop to make sure all requests are processed and synchronized
+	coll.grpcSrv.GracefulStop()
+
+	defer coll.traceService.mtx.Unlock()
+	coll.traceService.mtx.Lock()
+	return coll.traceService.data
 }
 
-func (coll *collector) MetricsExportRequest() *metricsExportRequest {
-	// Give some time for the gRPC server to process the request.
-	timeout := time.NewTimer(100 * time.Millisecond)
-	defer timeout.Stop()
-	select {
-	case got := <-coll.metricsService.requests:
-		return &got
-	case <-timeout.C:
-		return nil
-	}
+func (coll *collector) ExportedMetrics() *metricsExportRequest {
+	// stop to make sure all requests are processed and synchronized
+	coll.grpcSrv.GracefulStop()
```

Comment:
This means the collector becomes unusable when this or `ExportedTraces` is called. This seems like a sharp edge that will cause issues in future testing.

Is this graceful stop still needed? If it is do we need to update documentation somewhere to try and help preventing this from causing future issues?

Reply:
It should not be needed. I will double-check it and remove.

Reply:
after removal

```
$ go test -race -count=20
PASS
ok      github.com/signalfx/splunk-otel-go/distro       17.476s
```

and after reading the metrics SDK code I agree that it is not needed 😉

Reply:
part of https://github.com/signalfx/splunk-otel-go/pull/1848/commits/854a374398ef4a11b52fcaf03a1f9d6bfb9a63d1